google/cloud/bigtable/examples/data_snippets.cc

```diff
@@ -1022,58 +902,347 @@ void WriteConditionally(google::cloud::bigtable::Table table, int argc,
   (std::move(table));
 }
 
+using CommandType = std::function<void(std::vector<std::string> const& argv)>;
+using TableCommandType = std::function<void(google::cloud::bigtable::Table,
+                                            std::vector<std::string>)>;
+
+std::map<std::string, CommandType>::value_type MakeCommandEntry(
+    std::string const& name, std::vector<std::string> const& args,
+    TableCommandType function) {
+  auto command = [=](std::vector<std::string> argv) {
+    if (argv.size() != 3 + args.size()) {
+      std::ostringstream os;
+      os << name << " <project-id> <instance-id> <table-id>";
+      char const* sep = " ";
+      for (auto const& a : args) {
+        os << sep << a;
+      }
+      throw Usage{std::move(os).str()};
+    }
+    google::cloud::bigtable::Table table(
+        google::cloud::bigtable::CreateDefaultDataClient(
+            argv[0], argv[1], google::cloud::bigtable::ClientOptions()),
+        argv[2]);
+    argv.erase(argv.begin(), argv.begin() + 3);
+    function(table, argv);
+  };
+  return {name, command};
+}
+
+std::string DefaultTablePrefix() { return "tbl-data-"; }
+
+std::string TablePrefix(std::string const& prefix,
+                        std::chrono::system_clock::time_point tp) {
+  auto as_seconds =
+      std::chrono::duration_cast<std::chrono::seconds>(tp.time_since_epoch());
+  return prefix + std::to_string(as_seconds.count()) + "-";
+}
+
+std::string RandomTableId(std::string const& prefix,
+                          google::cloud::internal::DefaultPRNG& generator) {
+  return TablePrefix(prefix, std::chrono::system_clock::now()) +
+         google::cloud::internal::Sample(generator, 8,
+                                         "abcdefhijklmnopqrstuvwxyz");
+}
+
+void CleanupOldTables(std::string const& prefix,
+                      google::cloud::bigtable::TableAdmin admin) {
+  auto const threshold =
+      std::chrono::system_clock::now() - std::chrono::hours(48);
+  auto const max_table_name = TablePrefix(prefix, threshold);
+
+  namespace cbt = google::cloud::bigtable;
+  auto tables = admin.ListTables(cbt::TableAdmin::NAME_ONLY);
+  if (!tables) return;
+  for (auto const& t : *tables) {
+    if (t.name().rfind(prefix, 0) != 0) continue;
+    if (t.name() >= max_table_name) continue;
```

Comment:
Nit: This looks like it would break when the `as_seconds` field changes its width.  Admittedly not until 2286, but perhaps worth a comment.

Reply:
Done.

Reply:
Touché.